js/modules/loki_file_server_api.js

```diff
@@ -60,43 +53,47 @@ class LokiFileServerAPI {
 
   async verifyUserObjectDeviceMap(pubKeys, isRequest, iterator) {
     const users = await this.getDeviceMappingForUsers(pubKeys);
-    // log.info('verifyUserObjectDeviceMap Found', users.length, 'users')
 
     // go through each user and find deviceMap annotations
     const notFoundUsers = [];
-    await Promise.all(users.map(async user => {
-      let found = false;
-      if (!user.annotations || !user.annotations.length) {
-        log.info(
-          `verifyUserObjectDeviceMap no annotation for ${user.username}`
-        );
-        return;
-      }
-      const mappingNote = user.annotations.find(note => note.type === DEVICE_MAPPING_ANNOTATION_KEY);
-      const { authorisations } = mappingNote.value;
-      if (!Array.isArray(authorisations)) {
-        return;
-      }
-      await Promise.all(authorisations.map(async auth => {
-        // only skip, if in secondary search mode
-        if (isRequest && auth.secondaryDevicePubKey !== user.username) {
-          // this is not the authorization we're looking for
+    await Promise.all(
+      users.map(async user => {
+        let found = false;
+        if (!user.annotations || !user.annotations.length) {
           log.info(
-            `Request and ${auth.secondaryDevicePubKey} != ${user.username}`
+            `verifyUserObjectDeviceMap no annotation for ${user.username}`
           );
           return;
         }
-        const valid = await libloki.crypto.validateAuthorisation(auth);
-        // log.info('auth is valid for', user.username)
-        if (iterator(user.username, auth)) {
-          found = true;
+        const mappingNote = user.annotations.find(
+          note => note.type === DEVICE_MAPPING_ANNOTATION_KEY
+        );
+        const { authorisations } = mappingNote.value;
```

Comment:
mappingNote could potentially be falsy here right? Maybe return early if that's the case.